fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/api/event/player/BlockBreakEvents.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.player;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+public final class BlockBreakEvents {
+	private BlockBreakEvents() { }
+
+	/**
+	 * Callback before a block is broken.
+	 * Only called on the server, however updates are synced with the client.
+	 *
+	 * <p>Upon return:
+	 * <ul><li>SUCCESS/PASS passes on the callback to the next listener
+	 * <li>CONSUME continues the default code for breaking the block and ignores all other listeners
+	 * <li>FAIL cancels the block breaking action</ul>
```

Comment:
Why even make this event cancellable? We alreday have the attack block callback that provides a better UX for preventing the breaking of blocks.

I do see the need for a BlockBreakEvent that happens once the block is definitely going to be broken.

Reply:
Logging is one use, which is useful for more than just core protect

Reply:
Hmm, how about hack clients?

Reply:
Attack block is purely for UX, it doesn't actually prevent the breaking. As the JavaDoc on it says:
`ATTACK_BLOCK does not let you control the packet sending process yet.`

Reply:
Yes, it does allow you to stop the breaking of blocks. ive used it for this exact reason.